platform/src/main/java/org/hillview/table/ListComparator.java

```diff
@@ -19,19 +19,20 @@
 
 import org.hillview.table.api.IndexComparator;
 
+import java.util.ArrayList;
 import java.util.List;
 
 public class ListComparator extends IndexComparator {
-    private final List<IndexComparator> comparatorList;
+    private final ArrayList<IndexComparator> comparatorList;
```

Comment:
Why does this make a difference?

Reply:
Perhaps not the best way to do this, but I wanted it to be clear that we intend to directly index into the array when iterating through it. Otherwise, we create a lot of Iterator garbage.